rpc/QueryIterator.java

```diff
@@ -67,7 +74,12 @@ protected T computeNext() {
             case RES_NOT_SET:
                 throw new GraknClientException(MISSING_RESPONSE.message(className(TransactionProto.Transaction.Res.class)));
             default:
-                return transformResponse.apply(res);
+                currIterator = transformResponse.apply(res).iterator();
+                if (currIterator.hasNext()) {
+                    return currIterator.next();
+                } else {
+                    return computeNext();
```

Comment:
Although the server will not send response with no answer (it will just send `done` message in that case), I feel it's still nicer to have client handle this case by simply calling `computeNext`.